cpp/include/cudf/detail/distinct_hash_join.cuh

```diff
@@ -143,12 +116,37 @@ struct distinct_hash_join {
    */
   std::pair<std::unique_ptr<rmm::device_uvector<size_type>>,
             std::unique_ptr<rmm::device_uvector<size_type>>>
-  inner_join(rmm::cuda_stream_view stream, rmm::device_async_resource_ref mr) const;
+  inner_join(cudf::table_view const& probe,
+             rmm::cuda_stream_view stream,
+             rmm::device_async_resource_ref mr) const;
 
   /**
    * @copydoc cudf::distinct_hash_join::left_join
    */
   std::unique_ptr<rmm::device_uvector<size_type>> left_join(
-    rmm::cuda_stream_view stream, rmm::device_async_resource_ref mr) const;
+    cudf::table_view const& probe,
+    rmm::cuda_stream_view stream,
+    rmm::device_async_resource_ref mr) const;
+
+ private:
+  using probing_scheme_type = cuco::linear_probing<1, hasher>;
+  using cuco_storage_type   = cuco::storage<1>;
+
+  /// Hash table type
+  using hash_table_type = cuco::static_set<cuco::pair<hash_value_type, rhs_index_type>,
+                                           cuco::extent<size_type>,
+                                           cuda::thread_scope_device,
+                                           always_not_equal,
+                                           probing_scheme_type,
+                                           cudf::detail::cuco_allocator<char>,
+                                           cuco_storage_type>;
+
+  bool _has_nulls;           ///< True if nulls are present in either build table or probe table
```

Comment:
Since we do not specify the probe table, and we may not know anything about the probe table, should we set this always `true`?

Reply:
I'm following the same null check pattern used in `hash_join`, so I assume downstream users like Spark are handling this correctly. If needed, I'm happy to open a separate PR to update both `distinct_hash_join` and `hash_join` to improve this.

Reply:
Oh actually you reviewed that PR. Here it is: https://github.com/rapidsai/cudf/pull/13120.

The bug was fixed by checking `hash_nulls` on both table, like in https://github.com/rapidsai/cudf/pull/13120/files#diff-734dd746efe774e94501b6aa986e7824656182f85bf9af8bf30036c002ca1b82R82.

Reply:
I'm not sure if Spark can handle this correctly but assuming that we stumbled upon the same issue before, I would highly suspect that it couldn't do it.

Reply:
I think Spark handles this properly, e.g.: https://github.com/rapidsai/cudf/blob/e8935b9959dc4e62a1e486fb0359374df0e6e2ea/java/src/main/native/src/TableJni.cpp#L2904-L2906

Are you suggesting that we should remove `_has_null` as a data member as it's always `true`?

Reply:
Yes, I think so. Previously we have both tables thus we can compute `has_nulls` but now we only have one table thus this variable should be removed and the corresponding nullate value should always be `true`.

Reply:
done